app/verification.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+import requests
+
+from . import config, db, utils
+
+
+def update():
+    with db.redis_conn.pipeline() as p:
+        with requests.Session() as session:
+            for verdict in ["verified", "blocked"]:
+                r = session.get(
+                    f"https://raw.githubusercontent.com/flathub/backend/master/data/verification/{verdict}.json"
+                )
+                if r.status_code == 200:
+                    value = r.json()
+                    p.unlink(f"verification:{verdict}")
+                    if len(value):
+                        p.sadd(f"verification:{verdict}", *r.json())
+
+        p.execute()
+
+
+def initialize():
+    verification_dir = os.path.join(config.settings.datadir, "verification")
+    with db.redis_conn.pipeline() as p:
+        for verdict in ["verified", "blocked"]:
+            value = utils.get_appids(os.path.join(verification_dir, verdict + ".json"))
+            p.unlink(f"verification:{verdict}")
+            if len(value):
+                p.sadd(f"verification:{verdict}", *value)
+        p.execute()
+
+
+def get_verification_methods(appid: str):
+    segments = appid.split(".")
+
+    if len(segments) < 3:
+        return {
+            "methods": [],
+            "detail": "malformed_app_id",
+        }
+
+    elif db.redis_conn.sismember("verification:blocked", appid):
+        return {
+            "methods": [],
+            "detail": "blocked_by_admins",
+        }
+
+    elif segments[0:2] in [["com", "github"], ["io", "github"]]:
```

Comment:
I might be confused, but not sure (without trying) why this is `segments[0:2]` and not `segments[0:1]`

Reply:
The end index is exclusive.